phpunit/class-wp-rest-navigation-fallbacks-controller-test.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Unit tests covering WP_REST_Navigation_Fallbacks_Controller functionality.
+ *
+ * Note: that these tests are designed to provide high level coverage only. The majority of the tests
+ * are made directly against the WP_Navigation_Fallbacks_Gutenberg class as this:
+ *
+ * - is where the bulk of the logic is.
+ * - is also consumed by the Navigation block's server side rendering.
+ *
+ * @package WordPress
+ * @subpackage REST API
+ */
+
+/**
+ * @group restapi
+ * @group navigation
+ */
+class WP_REST_Navigation_Fallbacks_Controller_Test extends WP_Test_REST_Controller_Testcase {
+
+	protected static $admin_user;
+	protected static $editor_user;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$admin_user = $factory->user->create( array( 'role' => 'administrator' ) );
+
+		self::$editor_user = $factory->user->create( array( 'role' => 'editor' ) );
+	}
+
+	public function set_up() {
+		parent::set_up();
+
+		wp_set_current_user( self::$admin_user );
+	}
+
+	/**
+	 * @covers WP_REST_Navigation_Fallbacks_Controller::register_routes
+	 *
+	 * @since 5.8.0
+	 * @since 6.2.0 Added pattern directory categories endpoint.
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+
+		$this->assertArrayHasKey( '/wp-block-editor/v1/navigation-fallbacks', $routes );
+	}
+
+	public function test_should_not_return_menus_for_users_without_permissions() {
```

Comment:
Add these need covers notation.

Reply:
I wasn't entirely sure whether I needed to specify the `method` or whether I could just specify the REST class itself. I went with the class but I can update if required.